### PR TITLE
chore(schematics): Nx workspace dt-component schematics.

### LINKED ISSUE #889: [Schematics] Create dt-barista-component schematic
It should create a new folder inside the libs/barista-components using the '@nrwl/angular', 'library' schematic. This takes care of adding the references in the nx.json and angular.json file for the new library. 
All additional files or changes should follow the nrwl library schematic to conform to our structure. 
 
This schematic should also use the dt-example, dt-dev-page and dt-e2e schematic internally. 

The schematic should be part of the libs/workspace package.

### PR DESCRIPTION
**Pull request by bardosmisi**
### <strong>Pull Request</strong>

Dt-component schematics in NX workspace.

#### Type of PR

Other (if none of the above apply)

Closes #889

### CODE REVIEW

File: tools/schematics/dt-component/files/libs/barista-components/__name@dasherize__/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        include = ["**/*.ts"],
+        exclude = [
+            "**/*.spec.ts",
+            "src/test-setup.ts",
+        ],
+    ) + glob([
+        "**/*.html",
+        "**/*.scss",
+    ]),
+)
```
Comment (lukasholzer) [inline tools/schematics/dt-component/files/libs/barista-components/__name@dasherize__/BUILD.bazel:17]: I would add a small setup for typescript like the button.

File: tools/schematics/dt-component/files/libs/barista-components/__name@dasherize__/barista.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{
+  "title": "",
+  "description": "",
+  "postid": "<%= dasherize(name) %>",
+  "identifier": "",
+  "category": "components",
+  "public": false,
```
Comment (lukasholzer) [inline tools/schematics/dt-component/files/libs/barista-components/__name@dasherize__/barista.json:7]: why not public?

File: tools/schematics/dt-component/files/libs/barista-components/__name@dasherize__/jest.config.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+module.exports = {
```
Comment (lukasholzer) [inline tools/schematics/dt-component/files/libs/barista-components/__name@dasherize__/jest.config.js:1]: please create a `jest.config.json` like the button.

**Review by lukasholzer — CHANGES_REQUESTED** [review]
Can you leverage the nx create angular library schematic for creating all the files and then just update the files that we need?

File: tools/schematics/dt-component/files/libs/barista-components/__name@dasherize__/barista.json (hunk 2 above)
Comment (bardosmisi) [reply tools/schematics/dt-component/files/libs/barista-components/__name@dasherize__/barista.json]: I thought not public as the default would be a good idea. Fixed it though.

File: tools/schematics/dt-component/files/libs/barista-components/__name@dasherize__/BUILD.bazel (hunk 1 above)
Comment (bardosmisi) [reply tools/schematics/dt-component/files/libs/barista-components/__name@dasherize__/BUILD.bazel:17]: Added it, thanks.

File: tools/schematics/dt-component/files/libs/barista-components/__name@dasherize__/jest.config.js (hunk 3 above)
Comment (bardosmisi) [reply tools/schematics/dt-component/files/libs/barista-components/__name@dasherize__/jest.config.js:1]: Added it, does that mean I should remove the `jest.config.js`?

Comment (lukasholzer) [reply tools/schematics/dt-component/files/libs/barista-components/__name@dasherize__/jest.config.js:1]: yep

**Review by lukasholzer — CHANGES_REQUESTED** [review]
Use the existing `@nrwl/angular:library` schematic to create the library and update the existing files:

image

and then add the bazel files and update the barista specific stuff.

**Review by tomheller — APPROVED** [review]

**Review by tomheller — DISMISSED** [review]

**Review by ffriedl89 — APPROVED** [review]

**Review by lukasholzer — APPROVED** [review]

**Pull request merged**